.github/workflows/ci-linux.yml

```diff
@@ -0,0 +1,50 @@
+name: CI-linux
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  tests:
+    name: "Python ${{ matrix.python-version }}"
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        python-version: [2.7, 3.6, 3.7, 3.8]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "${{ matrix.python-version }}"
+
+      - name: Install dependencies
+        run: |
+          set -xe
+          python -VV
+          python -m site
+          pip install --upgrade pip setuptools wheel
+
+      - name: Install from source
+        run: pip install .
+
+      - name: Install dev requirements
+        run: pip install -r requirements-dev.txt
+
+      - name: Lint
+        run: flake8 src/h3 setup.py tests
+
+      - name: Tests
+        run: pytest --cov=h3 --full-trace --cov-report=xml
+
+      - name: Upload coverage to Codecov
```

Comment:
Looks like this is ungated, so all the Linux runs report to codecov - not sure that's intended rather than using a single run to generate the coverage

Reply:
I'm not actually sure what happens with the multiple reports to codecov, other than that it seems to work OK. Is there a downside to having multiple reports?

Reply:
I don't know what codecov does in that case.

Reply:
I'll leave it in for now, as we were doing the same thing with Travis CI, and since everything seems to work OK.

I'll revisit later, when I try to clean up and better organize the CI scripts.